lib/db_utils.go

```diff
@@ -244,50 +244,89 @@ type DBPrefixes struct {
+	// if you simply add a group member after the block height DeSoAccessGroupsBlockHeight.
 	//
-	// * Note that GroupMessagingPublicKey != GroupOwnerPublicKey. For this index
-	//   it was convenient for various reasons to put the messaging public key into
-	//   the index rather than the group owner's public key. This becomes clear if
-	//   you read all the fetching code around this index.
+	// <prefix, GroupMemberPublicKey [33]byte, GroupMessagingPublicKey [33]byte> -> <AccessGroupSimplifiedEntry>
 	//
-	// <prefix, OwnerPublicKey [33]byte, GroupMessagingPublicKey [33]byte> -> <HackedMessagingKeyEntry>
-	PrefixMessagingGroupMetadataByMemberPubKeyAndGroupMessagingPubKey []byte `prefix_id:"[58]" is_state:"true"`
+	// Deprecation comment: We are deprecating this key because there may be some edge cases in which
+	// two groups share the same GroupMessagingPublicKey and yet are actually different groups. The
+	// new index below allows us to distinguish between these by adding the GroupKeyName in the key.
+	DeprecatedPrefixGroupMembershipIndex []byte `prefix_id:"[58]" is_state:"true"` // Deprecated: Use PrefixGroupMembershipIndex instead
+
+	// This prefix is used to store all mappings for access group members. The group owner has a mapping
+	// with <groupOwnerPk, groupOwnerPk, groupName> and then everybody else has
+	// <memberPk, groupOwnerPk, groupName>. We don't need to store members in a group entry anymore since
+	// we can just iterate over the members in the group membership index here. This saves us a lot of space
+	// and makes it easier to add and remove members from groups.
+	//
+	// * Note that as mentioned above, there is a special case where GroupMemberPublicKey == GroupOwnerPublicKey.
+	//   For this index it was convenient for various reasons to automatically save an entry
+	//   with such key in the db whenever user registers a group. This becomes clear if
+	//   you read all the fetching code around this index. Particularly functions containing
+	//   the 'owner' keyword. This is not a bug, it's a feature because we might want an owner to be a member
+	//   of their own group for various reasons:
+	//   - To be able to read messages sent to the group if group was created with a derived key.
+	//   - To be able to fetch all groups that a user is a member of (including groups that
+	//     they own). This is especially useful for Backend API to fetch all groups for a user.
+	//
+	// New <GroupMembershipIndex> :
+	// <prefix, GroupMemberPublicKey [33]byte, GroupOwnerPublicKey [33]byte, GroupKeyName [32]byte> -> <AccessGroupMember>
+	PrefixGroupMembershipIndex []byte `prefix_id:"[63]" is_state:"true"`
+
+	// Prefix for Enumerating over all the members of a group:
+	// <prefix, GroupOwnerPublicKey [33]byte, GroupKeyName [32]byte, GroupMemberPublicKey [33]byte> -> <>
+	PrefixGroupEnumerationIndex []byte `prefix_id:"[64]" is_state:"true"`
```

Comment:
Can we call this `PrefixGroupMemberEnumerationIndex`?

Reply:
+1